Rubberduck.Parsing/Symbols/DeclarationFinder.cs

```diff
@@ -1084,10 +1089,24 @@ private IEnumerable<Declaration> GetNameCollisionDeclarations(IEnumerable<Identi
         }
 
         private bool IsAccessibleInOtherProcedureModule(Declaration candidate, Declaration declaration)
+            => AreProceduralModules(candidate, declaration, out bool isSameModule)
+                && !isSameModule && candidate.Accessibility != Accessibility.Private;
+
+        private bool IsAccessibleInSameProcedureModule(Declaration candidate, Declaration declaration)
```

Comment:
If this is going to be changing, it needs to consider performance. The check for `candidate.Accessibility != Accessibility.Private` is cheap. Really cheap compared to `AreProceduralModules`, so it needs to short circuit on that *first*. The existing code is already bad, if this is being changed it should be fixed.

Reply:
Agreed. I've re-implemented the conflict search to be far less 'expensive'.  It now looks for use of the proposed name (`DeclarationFinder.MatchName()`).  And, if it is found, then evaluate the matched name(s) to see if they represent a conflict with the proposed rename target.  As you've noted, the existing conflict search is very expensive. It creates a list of all possible conflicts for a given declaration before checking if the proposed name is ever used.

Also, the new method `FindNewDeclarationNameConflicts` looks more like a `DeclarationFinder` extension method to me.  Thoughts?